### PR TITLE
games-emulation/ppsspp: bump to 1.10.3; drop old; add proxy-maint

### PR DESCRIPTION
**Pull request by xgqt**
Package-Manager: Portage-3.0.9, Repoman-3.0.2
Signed-off-by: Maciej Barć <xgqt@protonmail.com>

add myself to games-emulation/ppsspp maintainers

### CODE REVIEW

**Comment by xgqt** [comment]
dx9sdk isn't needed for Linux builds afaik

**Review by juippis — COMMENTED** [review]
```
[ERROR] >>> Not all runs were successful.
atom: =games-emulation/ppsspp-1.10.3, USE flags: 'headless qt5 sdl system-ffmpeg'
atom: =games-emulation/ppsspp-1.10.3, USE flags: 'headless qt5 -sdl system-ffmpeg'
atom: =games-emulation/ppsspp-1.10.3, USE flags: 'headless qt5 sdl -system-ffmpeg'
atom: =games-emulation/ppsspp-1.10.3, USE flags: 'headless qt5 -sdl -system-ffmpeg'
atom: =games-emulation/ppsspp-1.10.3, USE flags: '-headless qt5 sdl system-ffmpeg'
atom: =games-emulation/ppsspp-1.10.3, USE flags: '-headless qt5 -sdl system-ffmpeg'
```
so it's pretty clear this isn't working with `qt5` USE flag enabled. Can you test?

And with `-qt5` I get lots of QA notices, 
```
 * QA Notice: Unresolved soname dependencies:
 * 
 * 	/usr/bin/PPSSPPSDL: libglslang.so
 * 
 * system-ffmpeg USE flag is enabled, some bugs might arise due to it.
 * See https://github.com/hrydgard/ppsspp/issues/9026 for more informations.
 * QA Notice: The following files contain insecure RUNPATHs
 *  Please file a bug about this at https://bugs.gentoo.org/
 *  with the maintainer of the package.
 *   /var/tmp/portage/gpm/portage/games-emulation/ppsspp-1.10.3/image/usr/bin/PPSSPPSDL
 *     RPATH: /var/tmp/portage/gpm/portage/games-emulation/ppsspp-1.10.3/work/ppsspp-1.10.3_build/lib
 * 
```
The first one is a missing dependency, the 2nd one is something that can be be patched with cmake build system. 

You can enable these settings with
https://wiki.gentoo.org/wiki/Package_testing#make.conf to test your package.

**Comment by xgqt** [comment]
>  * QA Notice: Unresolved soname dependencies:
>  * 
>  * 	/usr/bin/PPSSPPSDL: libglslang.so

`dev-util/glslang` covers that
Are you sure it is on your system?

**Comment by xgqt** [comment]
https://github.com/gentoo/gentoo/pull/18336/files#diff-394b0e7d413f985a5697eebdb9f6969aebe15be6d9105b4c6d016afe15bfdbefR32

**Comment by juippis** [comment]
I have it, 
```
 * Contents of dev-util/glslang-10.11.0.0_pre20200924:
/usr
/usr/bin
/usr/bin/glslangValidator
/usr/bin/spirv-remap
/usr/include
/usr/include/glslang
/usr/include/glslang/HLSL
/usr/include/glslang/HLSL/hlslAttributes.h
/usr/include/glslang/HLSL/hlslGrammar.h
/usr/include/glslang/HLSL/hlslOpMap.h
/usr/include/glslang/HLSL/hlslParseHelper.h
/usr/include/glslang/HLSL/hlslParseables.h
/usr/include/glslang/HLSL/hlslScanContext.h
/usr/include/glslang/HLSL/hlslTokenStream.h
/usr/include/glslang/HLSL/hlslTokens.h
/usr/include/glslang/Include
/usr/include/glslang/Include/BaseTypes.h
/usr/include/glslang/Include/Common.h
/usr/include/glslang/Include/ConstantUnion.h
/usr/include/glslang/Include/InfoSink.h
/usr/include/glslang/Include/InitializeGlobals.h
/usr/include/glslang/Include/PoolAlloc.h
/usr/include/glslang/Include/ResourceLimits.h
/usr/include/glslang/Include/ShHandle.h
/usr/include/glslang/Include/Types.h
/usr/include/glslang/Include/arrays.h
/usr/include/glslang/Include/glslang_c_interface.h
/usr/include/glslang/Include/glslang_c_shader_types.h
/usr/include/glslang/Include/intermediate.h
/usr/include/glslang/MachineIndependent
/usr/include/glslang/MachineIndependent/Initialize.h
/usr/include/glslang/MachineIndependent/LiveTraverser.h
/usr/include/glslang/MachineIndependent/ParseHelper.h
/usr/include/glslang/MachineIndependent/RemoveTree.h
/usr/include/glslang/MachineIndependent/Scan.h
/usr/include/glslang/MachineIndependent/ScanContext.h
/usr/include/glslang/MachineIndependent/SymbolTable.h
/usr/include/glslang/MachineIndependent/Versions.h
/usr/include/glslang/MachineIndependent/attribute.h
/usr/include/glslang/MachineIndependent/gl_types.h
/usr/include/glslang/MachineIndependent/glslang_tab.cpp.h
/usr/include/glslang/MachineIndependent/iomapper.h
/usr/include/glslang/MachineIndependent/localintermediate.h
/usr/include/glslang/MachineIndependent/parseVersions.h
/usr/include/glslang/MachineIndependent/preprocessor
/usr/include/glslang/MachineIndependent/preprocessor/PpContext.h
/usr/include/glslang/MachineIndependent/preprocessor/PpTokens.h
/usr/include/glslang/MachineIndependent/propagateNoContraction.h
/usr/include/glslang/MachineIndependent/reflection.h
/usr/include/glslang/Public
/usr/include/glslang/Public/ShaderLang.h
/usr/include/glslang/SPIRV
/usr/include/glslang/SPIRV/GLSL.ext.AMD.h
/usr/include/glslang/SPIRV/GLSL.ext.EXT.h
/usr/include/glslang/SPIRV/GLSL.ext.KHR.h
/usr/include/glslang/SPIRV/GLSL.ext.NV.h
/usr/include/glslang/SPIRV/GLSL.std.450.h
/usr/include/glslang/SPIRV/GlslangToSpv.h
/usr/include/glslang/SPIRV/Logger.h
/usr/include/glslang/SPIRV/NonSemanticDebugPrintf.h
/usr/include/glslang/SPIRV/SPVRemapper.h
/usr/include/glslang/SPIRV/SpvBuilder.h
/usr/include/glslang/SPIRV/SpvTools.h
/usr/include/glslang/SPIRV/bitutils.h
/usr/include/glslang/SPIRV/disassemble.h
/usr/include/glslang/SPIRV/doc.h
/usr/include/glslang/SPIRV/hex_float.h
/usr/include/glslang/SPIRV/spirv.hpp
/usr/include/glslang/SPIRV/spvIR.h
/usr/include/glslang/build_info.h
/usr/lib
/usr/lib/cmake
/usr/lib/cmake/HLSLTargets-gentoo.cmake
/usr/lib/cmake/HLSLTargets.cmake
/usr/lib/cmake/OGLCompilerTargets-gentoo.cmake
/usr/lib/cmake/OGLCompilerTargets.cmake
/usr/lib/cmake/OSDependentTargets-gentoo.cmake
/usr/lib/cmake/OSDependentTargets.cmake
/usr/lib/cmake/SPIRVTargets-gentoo.cmake
/usr/lib/cmake/SPIRVTargets.cmake
/usr/lib/cmake/SPVRemapperTargets-gentoo.cmake
/usr/lib/cmake/SPVRemapperTargets.cmake
/usr/lib/cmake/glslang-default-resource-limitsTargets-gentoo.cmake
/usr/lib/cmake/glslang-default-resource-limitsTargets.cmake
/usr/lib/cmake/glslangTargets-gentoo.cmake
/usr/lib/cmake/glslangTargets.cmake
/usr/lib/cmake/glslangValidatorTargets-gentoo.cmake
/usr/lib/cmake/glslangValidatorTargets.cmake
/usr/lib/cmake/spirv-remapTargets-gentoo.cmake
/usr/lib/cmake/spirv-remapTargets.cmake
/usr/lib/libHLSL.so
/usr/lib/libOGLCompiler.a
/usr/lib/libOSDependent.a
/usr/lib/libSPIRV.so
/usr/lib/libSPVRemapper.so
/usr/lib/libglslang-default-resource-limits.so
/usr/lib/libglslang.so -> libglslang.so.11
/usr/lib/libglslang.so.11 -> libglslang.so.11.0.0
/usr/lib/libglslang.so.11.0.0
/usr/lib64
/usr/lib64/cmake
/usr/lib64/cmake/HLSLTargets-gentoo.cmake
/usr/lib64/cmake/HLSLTargets.cmake
/usr/lib64/cmake/OGLCompilerTargets-gentoo.cmake
/usr/lib64/cmake/OGLCompilerTargets.cmake
/usr/lib64/cmake/OSDependentTargets-gentoo.cmake
/usr/lib64/cmake/OSDependentTargets.cmake
/usr/lib64/cmake/SPIRVTargets-gentoo.cmake
/usr/lib64/cmake/SPIRVTargets.cmake
/usr/lib64/cmake/SPVRemapperTargets-gentoo.cmake
/usr/lib64/cmake/SPVRemapperTargets.cmake
/usr/lib64/cmake/glslang-default-resource-limitsTargets-gentoo.cmake
/usr/lib64/cmake/glslang-default-resource-limitsTargets.cmake
/usr/lib64/cmake/glslangTargets-gentoo.cmake
/usr/lib64/cmake/glslangTargets.cmake
/usr/lib64/cmake/glslangValidatorTargets-gentoo.cmake
/usr/lib64/cmake/glslangValidatorTargets.cmake
/usr/lib64/cmake/spirv-remapTargets-gentoo.cmake
/usr/lib64/cmake/spirv-remapTargets.cmake
/usr/lib64/libHLSL.so
/usr/lib64/libOGLCompiler.a
/usr/lib64/libOSDependent.a
/usr/lib64/libSPIRV.so
/usr/lib64/libSPVRemapper.so
/usr/lib64/libglslang-default-resource-limits.so
/usr/lib64/libglslang.so -> libglslang.so.11
/usr/lib64/libglslang.so.11 -> libglslang.so.11.0.0
/usr/lib64/libglslang.so.11.0.0
/usr/share
/usr/share/doc
/usr/share/doc/glslang-10.11.0.0_pre20200924
/usr/share/doc/glslang-10.11.0.0_pre20200924/README-spirv-remap.txt.bz2
/usr/share/doc/glslang-10.11.0.0_pre20200924/README.md.bz2
```

I believe it's related to the rpath issue. You'll need to read the build.log carefully ,especially anything related to rpath.

**Comment by xgqt** [comment]
> I have it,
> 
> ```
>  * Contents of dev-util/glslang-10.11.0.0_pre20200924:
> /usr
> /usr/bin
> /usr/bin/glslangValidator
> /usr/bin/spirv-remap
> /usr/include
> /usr/include/glslang
> /usr/include/glslang/HLSL
> /usr/include/glslang/HLSL/hlslAttributes.h
> /usr/include/glslang/HLSL/hlslGrammar.h
> /usr/include/glslang/HLSL/hlslOpMap.h
> /usr/include/glslang/HLSL/hlslParseHelper.h
> /usr/include/glslang/HLSL/hlslParseables.h
> /usr/include/glslang/HLSL/hlslScanContext.h
> /usr/include/glslang/HLSL/hlslTokenStream.h
> /usr/include/glslang/HLSL/hlslTokens.h
> /usr/include/glslang/Include
> /usr/include/glslang/Include/BaseTypes.h
> /usr/include/glslang/Include/Common.h
> /usr/include/glslang/Include/ConstantUnion.h
> /usr/include/glslang/Include/InfoSink.h
> /usr/include/glslang/Include/InitializeGlobals.h
> /usr/include/glslang/Include/PoolAlloc.h
> /usr/include/glslang/Include/ResourceLimits.h
> /usr/include/glslang/Include/ShHandle.h
> /usr/include/glslang/Include/Types.h
> /usr/include/glslang/Include/arrays.h
> /usr/include/glslang/Include/glslang_c_interface.h
> /usr/include/glslang/Include/glslang_c_shader_types.h
> /usr/include/glslang/Include/intermediate.h
> /usr/include/glslang/MachineIndependent
> /usr/include/glslang/MachineIndependent/Initialize.h
> /usr/include/glslang/MachineIndependent/LiveTraverser.h
> /usr/include/glslang/MachineIndependent/ParseHelper.h
> /usr/include/glslang/MachineIndependent/RemoveTree.h
> /usr/include/glslang/MachineIndependent/Scan.h
> /usr/include/glslang/MachineIndependent/ScanContext.h
> /usr/include/glslang/MachineIndependent/SymbolTable.h
> /usr/include/glslang/MachineIndependent/Versions.h
> /usr/include/glslang/MachineIndependent/attribute.h
> /usr/include/glslang/MachineIndependent/gl_types.h
> /usr/include/glslang/MachineIndependent/glslang_tab.cpp.h
> /usr/include/glslang/MachineIndependent/iomapper.h
> /usr/include/glslang/MachineIndependent/localintermediate.h
> /usr/include/glslang/MachineIndependent/parseVersions.h
> /usr/include/glslang/MachineIndependent/preprocessor
> /usr/include/glslang/MachineIndependent/preprocessor/PpContext.h
> /usr/include/glslang/MachineIndependent/preprocessor/PpTokens.h
> /usr/include/glslang/MachineIndependent/propagateNoContraction.h
> /usr/include/glslang/MachineIndependent/reflection.h
> /usr/include/glslang/Public
> /usr/include/glslang/Public/ShaderLang.h
> /usr/include/glslang/SPIRV
> /usr/include/glslang/SPIRV/GLSL.ext.AMD.h
> /usr/include/glslang/SPIRV/GLSL.ext.EXT.h
> /usr/include/glslang/SPIRV/GLSL.ext.KHR.h
> /usr/include/glslang/SPIRV/GLSL.ext.NV.h
> /usr/include/glslang/SPIRV/GLSL.std.450.h
> /usr/include/glslang/SPIRV/GlslangToSpv.h
> /usr/include/glslang/SPIRV/Logger.h
> /usr/include/glslang/SPIRV/NonSemanticDebugPrintf.h
> /usr/include/glslang/SPIRV/SPVRemapper.h
> /usr/include/glslang/SPIRV/SpvBuilder.h
> /usr/include/glslang/SPIRV/SpvTools.h
> /usr/include/glslang/SPIRV/bitutils.h
> /usr/include/glslang/SPIRV/disassemble.h
> /usr/include/glslang/SPIRV/doc.h
> /usr/include/glslang/SPIRV/hex_float.h
> /usr/include/glslang/SPIRV/spirv.hpp
> /usr/include/glslang/SPIRV/spvIR.h
> /usr/include/glslang/build_info.h
> /usr/lib
> /usr/lib/cmake
> /usr/lib/cmake/HLSLTargets-gentoo.cmake
> /usr/lib/cmake/HLSLTargets.cmake
> /usr/lib/cmake/OGLCompilerTargets-gentoo.cmake
> /usr/lib/cmake/OGLCompilerTargets.cmake
> /usr/lib/cmake/OSDependentTargets-gentoo.cmake
> /usr/lib/cmake/OSDependentTargets.cmake
> /usr/lib/cmake/SPIRVTargets-gentoo.cmake
> /usr/lib/cmake/SPIRVTargets.cmake
> /usr/lib/cmake/SPVRemapperTargets-gentoo.cmake
> /usr/lib/cmake/SPVRemapperTargets.cmake
> /usr/lib/cmake/glslang-default-resource-limitsTargets-gentoo.cmake
> /usr/lib/cmake/glslang-default-resource-limitsTargets.cmake
> /usr/lib/cmake/glslangTargets-gentoo.cmake
> /usr/lib/cmake/glslangTargets.cmake
> /usr/lib/cmake/glslangValidatorTargets-gentoo.cmake
> /usr/lib/cmake/glslangValidatorTargets.cmake
> /usr/lib/cmake/spirv-remapTargets-gentoo.cmake
> /usr/lib/cmake/spirv-remapTargets.cmake
> /usr/lib/libHLSL.so
> /usr/lib/libOGLCompiler.a
> /usr/lib/libOSDependent.a
> /usr/lib/libSPIRV.so
> /usr/lib/libSPVRemapper.so
> /usr/lib/libglslang-default-resource-limits.so
> /usr/lib/libglslang.so -> libglslang.so.11
> /usr/lib/libglslang.so.11 -> libglslang.so.11.0.0
> /usr/lib/libglslang.so.11.0.0
> /usr/lib64
> /usr/lib64/cmake
> /usr/lib64/cmake/HLSLTargets-gentoo.cmake
> /usr/lib64/cmake/HLSLTargets.cmake
> /usr/lib64/cmake/OGLCompilerTargets-gentoo.cmake
> /usr/lib64/cmake/OGLCompilerTargets.cmake
> /usr/lib64/cmake/OSDependentTargets-gentoo.cmake
> /usr/lib64/cmake/OSDependentTargets.cmake
> /usr/lib64/cmake/SPIRVTargets-gentoo.cmake
> /usr/lib64/cmake/SPIRVTargets.cmake
> /usr/lib64/cmake/SPVRemapperTargets-gentoo.cmake
> /usr/lib64/cmake/SPVRemapperTargets.cmake
> /usr/lib64/cmake/glslang-default-resource-limitsTargets-gentoo.cmake
> /usr/lib64/cmake/glslang-default-resource-limitsTargets.cmake
> /usr/lib64/cmake/glslangTargets-gentoo.cmake
> /usr/lib64/cmake/glslangTargets.cmake
> /usr/lib64/cmake/glslangValidatorTargets-gentoo.cmake
> /usr/lib64/cmake/glslangValidatorTargets.cmake
> /usr/lib64/cmake/spirv-remapTargets-gentoo.cmake
> /usr/lib64/cmake/spirv-remapTargets.cmake
> /usr/lib64/libHLSL.so
> /usr/lib64/libOGLCompiler.a
> /usr/lib64/libOSDependent.a
> /usr/lib64/libSPIRV.so
> /usr/lib64/libSPVRemapper.so
> /usr/lib64/libglslang-default-resource-limits.so
> /usr/lib64/libglslang.so -> libglslang.so.11
> /usr/lib64/libglslang.so.11 -> libglslang.so.11.0.0
> /usr/lib64/libglslang.so.11.0.0
> /usr/share
> /usr/share/doc
> /usr/share/doc/glslang-10.11.0.0_pre20200924
> /usr/share/doc/glslang-10.11.0.0_pre20200924/README-spirv-remap.txt.bz2
> /usr/share/doc/glslang-10.11.0.0_pre20200924/README.md.bz2
> ```
> 
> I believe it's related to the rpath issue. You'll need to read the build.log carefully ,especially anything related to rpath.

So why does it say `Unresolved soname dependencies`

**Comment by xgqt** [comment]
[ppsspp.log](https://github.com/gentoo/gentoo/files/5584148/ppsspp.log)

**Comment by xgqt** [comment]
[PPSSPPSDL_run.log](https://github.com/gentoo/gentoo/files/5584161/PPSSPPSDL_run.log)

**Comment by xgqt** [comment]
Build log, Qt5 version:
[ppsspp_qt5.log](https://github.com/gentoo/gentoo/files/5584469/ppsspp_qt5.log)

**Comment by xgqt** [comment]
ppsspp qt5 version run log:
[PPSSPPSQt5_run.log](https://github.com/gentoo/gentoo/files/5584475/PPSSPPSQt5_run.log)

File: games-emulation/ppsspp/ppsspp-1.10.3.ebuild [hunk 1]
```diff
@@ -1,59 +1,66 @@
...
 	media-libs/glew:=
 	sys-libs/zlib:=
 	virtual/opengl
+	sdl? ( media-libs/libsdl2 )
 	qt5? (
 		dev-qt/qtcore:5
 		dev-qt/qtgui:5
 		dev-qt/qtopengl:5
 		dev-qt/qtwidgets:5
 		!sdl? ( dev-qt/qtmultimedia:5 )
```
Comment (juippis) [inline games-emulation/ppsspp/ppsspp-1.10.3.ebuild:42]:
```
-- Configuring done
CMake Error at CMakeLists.txt:920 (add_library):
  Target "native" links to target "Qt5::Multimedia" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at CMakeLists.txt:1962 (add_executable):
  Target "PPSSPPHeadless" links to target "Qt5::Multimedia" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:1437 (add_library):
  Target "Core" links to target "Qt5::Multimedia" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at CMakeLists.txt:2036 (add_executable):
  Target "PPSSPPQt" links to target "Qt5::Multimedia" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
 * ERROR: games-emulation/ppsspp-1.10.3::testworld failed (configure phase):
 *   cmake failed
```

File: games-emulation/ppsspp/ppsspp-1.10.3.ebuild [hunk 2]
```diff
@@ -66,32 +73,39 @@ src_prepare() {
...
+	cmake_src_prepare
 }
 
 src_configure() {
 	local mycmakeargs=(
+		$(cmake_use_find_package sdl SDL2)
 		-DHEADLESS=$(usex headless)
-		-DUSING_QT_UI=$(usex qt5)
-		$(cmake-utils_use_find_package sdl SDL2)
 		-DUSE_SYSTEM_FFMPEG=$(usex system-ffmpeg)
```
Comment (juippis) [inline games-emulation/ppsspp/ppsspp-1.10.3.ebuild:84]:
```
CMake Error at /usr/share/cmake/Modules/FindPkgConfig.cmake:545 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPkgConfig.cmake:733 (_pkg_check_modules_internal)
  cmake/Modules/FindFFmpeg.cmake:98 (pkg_check_modules)
  cmake/Modules/FindFFmpeg.cmake:120 (find_ffmpeg)
  CMakeLists.txt:539 (find_package)


-- Configuring incomplete, errors occurred!
See also "/var/tmp/portage/gpm/portage/games-emulation/ppsspp-1.10.3/work/ppsspp-1.10.3_build/CMakeFiles/CMakeOutput.log".
See also "/var/tmp/portage/gpm/portage/games-emulation/ppsspp-1.10.3/work/ppsspp-1.10.3_build/CMakeFiles/CMakeError.log".
 * ERROR: games-emulation/ppsspp-1.10.3::testworld failed (configure phase):
 *   cmake failed
```

**Review by juippis — COMMENTED** [review]
https://dev.gentoo.org/~juippis/logs/ppsspp-1.10.3-build.log

**Comment by juippis** [comment]
BTW I can see from your build.log that you're running a stable version. Please set up a chroot/container environment with ~ and try again. It could very well be related, I'm running latest available gcc, Qt (~) etc.

**Review by juippis — COMMENTED** [review]
Okay I emerged the Qt stack with `-gles2-only` and this worked. After doing `+gles2-only`, 
```
[ebuild   R    ] dev-qt/qtgui-5.15.2:5/5.15.2::gentoo  USE="X dbus egl gif gles2-only* jpeg libinput png udev -accessibility -debug -eglfs -evdev -ibus -test -tslib -tuio -vnc -vulkan -wayland" 0 KiB
[ebuild   R    ] dev-qt/qtwidgets-5.15.2:5/5.15::gentoo  USE="X gles2-only* gtk png -debug -test" 0 KiB
[ebuild   R    ] dev-qt/qtopengl-5.15.2:5/5.15::gentoo  USE="gles2-only* -debug -test" 0 KiB
[ebuild   R    ] dev-qt/qtprintsupport-5.15.2:5/5.15::gentoo  USE="cups gles2-only* -debug -test" 0 KiB
[ebuild   R    ] dev-qt/qtmultimedia-5.15.2:5/5.15::gentoo  USE="alsa gles2-only* widgets -debug -gstreamer -openal -pulseaudio -qml -test" 0 KiB
```
it's broken again. This needs to be patched, or dealt in ebuild somehow, since for me `+gles2-only` seems to be a profile default.

@a17r @chiitoo any ideas where this is coming from? And how to sophisticately solve it?

**Comment by a17r** [comment]
> since for me `+gles2-only` seems to be a profile default.

certainly not, check your `package.use`.

**Comment by juippis** [comment]
So is it fine to force `gles2-only` off from Qt deps in here, or can it easily be patched?

**Comment by a17r** [comment]
There's plenty of precedent:
```
./dev-qt/qtwayland/qtwayland-5.15.2.ebuild:             ~dev-qt/qtgui-${PV}[-gles2-only]
./dev-qt/qtwayland/qtwayland-5.15.1.ebuild:             ~dev-qt/qtgui-${PV}[-gles2-only]
./media-sound/fmit/fmit-1.0.15-r3.ebuild:       dev-qt/qtgui:5[-gles2-only]
./media-sound/qmmp/qmmp-1.4.2.ebuild:           dev-qt/qtgui:5[-gles2-only]
./media-sound/qmmp/qmmp-9999.ebuild:            dev-qt/qtgui:5[-gles2-only]
./games-fps/doomsday/doomsday-2.1.1-r1.ebuild:  dev-qt/qtgui:5=[-gles2-only]
./kde-apps/analitza/analitza-20.08.3.ebuild:    >=dev-qt/qtgui-${QTMIN}:5[-gles2-only]
./app-office/scribus/scribus-9999.ebuild:       dev-qt/qtgui:5[-gles2-only]
./app-office/scribus/scribus-1.5.6.1.ebuild:    dev-qt/qtgui:5[-gles2-only]
./app-office/scribus/scribus-1.5.5_p20200626.ebuild:    dev-qt/qtgui:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.0-r3.ebuild:          dev-qt/qtgui:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.0-r3.ebuild:          dev-qt/qtgui:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.0-r3.ebuild:          dev-qt/qtopengl:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.1.ebuild:             dev-qt/qtgui:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.1.ebuild:             dev-qt/qtgui:5[-gles2-only]
./sci-visualization/paraview/paraview-5.8.1.ebuild:             dev-qt/qtopengl:5[-gles2-only]
./dev-util/apitrace/apitrace-9.0-r1.ebuild:             dev-qt/qtgui:5[-gles2-only]
./dev-util/apitrace/apitrace-9.0-r1.ebuild:             dev-qt/qtwidgets:5[-gles2-only]
./media-gfx/openscad/openscad-2019.05-r3.ebuild:        dev-qt/qtgui:5[-gles2-only]
./media-gfx/openscad/openscad-9999.ebuild:      dev-qt/qtgui:5[-gles2-only]
./media-gfx/krita/krita-4.4.1.ebuild:   >=dev-qt/qtgui-${QTMIN}:5=[-gles2-only]
./media-gfx/krita/krita-4.3.0.ebuild:   >=dev-qt/qtgui-${QTMIN}:5=[-gles2-only]
./media-gfx/digikam/digikam-7.1.0.ebuild:       >=dev-qt/qtgui-${QTMIN}:5[-gles2-only]
```

**Comment by a17r** [comment]
@juippis that said, `./dev-libs/efl/efl-1.25.1-r10.ebuild:   wayland? ( gles2-only !opengl )` is a curious outlier that should probably be fixed.

**Comment by juippis** [comment]
@xgqt so just force `-gles2-only` where applicable, you can see examples in the list above.

@a17r I'd do it if I knew how, as it's an upstream requirement. I see it's buildable in meson, was probably prohibited with autotools.
https://git.enlightenment.org/core/efl.git/tree/configure.ac?h=v1.22.6#n2619

That requirement has given me headache. They only care about binary distributions and are deprecating `opengl` configure support altogether. 
```
18:49  juippis: raster: does wayland work with opengl=full? I see it compiles just fine
18:49  juippis: is there something in runtime
18:50  juippis: why do I remember it required opengl=es-egl. Did meson change this?
18:50 @raster: juippis:  no it doesnt
18:51 @raster: no gl accel will work unless u use egl+gles
18:51 @raster: gles is the default for wefl for various reasons
18:51 @raster: 1. its now available on all desktop drivers anyway
18:51 @raster: 2. some mobile/embedded drviers ONLY offer egl/gles
18:52 @raster: 3. our code only has support for egl in combo with gles. in theory we can use egl + full gl. our code doesnt
18:52 @raster: but there is no point as we dont need gull dedktop gl features
18:52 @raster: and we already have egl+gles
18:52 @raster: so its just busy work to do anything on that code-wise
```
I may need to revisit this. Some note about missing feature, making it experimental or outright disabling `opengl` from it.
But this is offtopic for this PR.

File: games-emulation/ppsspp/ppsspp-1.10.3.ebuild [hunk 3]
```diff
@@ -1,59 +1,67 @@
...
+	!system-ffmpeg? ( https://github.com/hrydgard/ppsspp-ffmpeg/archive/d3e695286cce346274192c214c00f021c00cd575.tar.gz -> ${P}-ffmpeg.tar.gz )
 "
 
+RESTRICT="test"
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~x86"
-IUSE="headless +qt5 sdl +system-ffmpeg"
+KEYWORDS="~amd64"
+IUSE="+headless qt5 sdl system-ffmpeg"
```
Comment (a17r) [inline games-emulation/ppsspp/ppsspp-1.10.3.ebuild:26]: What's the reason for flipping those defaults? Don't we want system libs used wherever possible? Could someone be surprised by `+headless` default?

I would expect an explanation inside git summary.

Comment (xgqt) [reply games-emulation/ppsspp/ppsspp-1.10.3.ebuild:26]: @a17r upstream doesn't really support building with system ffmpeg.
As for headless I think it's a good addition that doesn't really require much (look at the deps).
My reasoning is similar to snes9x where (yes - its not a flag but) you (by default) get a headless version.

Comment (a17r) [reply games-emulation/ppsspp/ppsspp-1.10.3.ebuild:26]: I see it does not even cause any additional dependencies. If it does not cause a measurable uptick in build image size I'd argue it shouldn't even be an `IUSE`, but default ON certainly doesn't hurt here.

Re system ffmpeg, upstreams don't always share goals with distributions, doesn't mean we easily give in, unless the option is known to be problematic.

Comment (xgqt) [reply games-emulation/ppsspp/ppsspp-1.10.3.ebuild:26]: > Re system ffmpeg, upstreams don't always share goals with distributions, doesn't mean we easily give in, unless the option is known to be problematic.

True, I myself run with all system-* on. Will switch to +.

**Review by juippis — APPROVED** [review]

**Comment by xgqt** [comment]
@juippis @a17r thanks for cooperation

**Comment by juippis** [comment]
Thanks for your interest and contributions :)

**Pull request closed without merging**